src/checkbox/checkbox.ts

```diff
@@ -0,0 +1,102 @@
+import {Component, ElementRef, EventEmitter, forwardRef, Input, Output} from '@angular/core';
+import {ControlValueAccessor, NG_VALUE_ACCESSOR} from '@angular/forms';
+
+const NGB_CHECKBOX_VALUE_ACCESSOR = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => NgbCheckbox),
+  multi: true
+};
+
+let checkbox_id_generator = 1;
+
+/**
+ * A component that allows the user to edit boolean values.
+ */
+@Component({
+  selector: 'ngb-checkbox',
+  host: {'[class.custom-control]': 'true', '[class.custom-checkbox]': 'true'},
```

Comment:
```ts
host: {'class': 'custom-control custom-checkbox'},
```
probably would be cleaner, shorter. As it never changes it does not require bracket notation for bindings.